### PR TITLE
Replace `fs` with `fs-extra` in `blocks.ts`

### PR DESCRIPTION
**Pull request by kachkaev**
This PR was created in response to https://github.com/vercel/nft/issues/293#issuecomment-1152330914. It follows #313 and relates to #215.

I bumped Next to 12.2 as it includes https://github.com/vercel/nft/pull/285 (it allows us to `node:path` instead of `path` and still get correct file detection).

### CODE REVIEW

File: site/src/lib/blocks.ts [hunk 1]
```diff
@@ -1,8 +1,9 @@
 import { BlockMetadata, BlockMetadataRepository } from "blockprotocol";
-// eslint-disable-next-line no-restricted-imports,unicorn/prefer-node-protocol -- https://github.com/vercel/nft/issues/293
-import fs from "fs";
-import glob from "glob";
+import fs from "fs-extra";
+import { globby } from "globby";
 import hostedGitInfo from "hosted-git-info";
+// eslint-disable-next-line unicorn/prefer-node-protocol -- Replace "path" with "node:path" after upgrading next from 12.1.6 https://github.com/vercel/nft/issues/293
+import path from "path";
```
Comment (kachkaev) [inline site/src/lib/blocks.ts:6]: A newer version of Next should include an updated version of `@vercel/nft`, which should allow us to restore `node:` prefix. I can keep this PR open till then. Context: https://github.com/vercel/nft/pull/285

File: site/src/lib/blocks.ts [hunk 2]
```diff
@@ -84,49 +85,56 @@ const getRepositoryUrl = (
...
-          .split("/")
-          .join("/blocks/")}`,
-        lastUpdated: null, // TODO: derive from block data when provided by the hub
-      };
+  const blockMetadataFilePaths = await globby(
+    path.resolve(process.cwd(), `public/blocks/**/block-metadata.json`),
+  );
+
+  const result: ExpandedBlockMetadata[] = [];
+  for (const blockMetadataFilePath of blockMetadataFilePaths) {
```
Comment (kachkaev) [inline site/src/lib/blocks.ts:93]: Most of the diff is related to whitespace. I switched from `.map()` to `for of` to be able to use `await` in the loop.

File: site/src/lib/blocks.ts (hunk 1 above)
Comment (kachkaev) [reply site/src/lib/blocks.ts]: This problem is no longer present in Next `12.1.7-canary.44`. I suggest waiting for a stable release of Next.js before finishing this PR.

File: site/src/lib/blocks.ts [hunk 3]
```diff
@@ -51,12 +51,12 @@ const generateBlockFileUrl = (
 const getRepositoryUrl = (
   repository: BlockMetadataRepository | undefined,
   commit: string,
-  path: string | undefined,
+  hubInfoDirPath: string | undefined,
```
Comment (kachkaev) [inline site/src/lib/blocks.ts:54]: Avoids shadowing of `path`

**Review by CiaranMn — APPROVED** [review]
Looks good, tested and blocks working on hub

**Pull request merged**